kuma/javascript/src/toc.jsx

```diff
@@ -1,12 +1,44 @@
 // @flow
 import * as React from 'react';
+import { useContext, useEffect } from 'react';
+
+import GAProvider from './ga-provider.jsx';
 import { gettext } from './l10n.js';
 
 type Props = {
     html: string
 };
 
 export default function TOC({ html }: Props) {
+    const ga = useContext(GAProvider.context);
+
+    /**
+     * Send a signal to GA when the user clicks on one of links
+     * in the table of contents
+     * @param {Object} event - The event object that was triggered
+     */
+    function sendTOCClicks(event: MouseEvent) {
+        if (event.target instanceof HTMLAnchorElement) {
+            const action = event.target.textContent;
+            const label = new URL(event.target.href).hash;
+
+            ga('send', {
+                hitType: 'event',
+                eventCategory: 'MozMenu',
+                eventAction: action,
+                eventLabel: label
+            });
+        }
+    }
+
+    useEffect(() => {
+        const documentToc = document.querySelector('.document-toc');
```

Comment:
You should only use `.querySelector` on stuff you don't have access to in the JSX. 
I demonstrated this here: https://codesandbox.io/s/proud-resonance-i542y

Reply:
Dang it, you are correct. I can totally do this using a `ref` on the section. Meh. Thanks, @peterbe